### ISSUE TITLE
[crash][vectorization] VPValue is not a live-in; it is defined by a VPDef inside a VPlan.

### ISSUE THREAD

**Issue by dbakunevich** [llvm/llvm-project#55334]
This bug was found by Azul FuzzGen fuzzer for CPP.

Godbolt link: https://godbolt.org/z/cY1YzEaqG

Run clang -O3 (x86_64) on the test
[crash_5.cpp.txt](https://github.com/llvm/llvm-project/files/8646920/crash_5.cpp.txt)

Passed on clang 15, so appears to be fresh.
Can't reproduce this failure on versions 14 and below.

Failure looks like:
```
clang-15: /home/dbakunevich/work/llvm-project/llvm-project/llvm/lib/Transforms/Vectorize/VPlanValue.h:189: llvm::Value* llvm::VPValue::getLiveInIRValue(): Assertion `!getDef() && "VPValue is not a live-in; it is defined by a VPDef inside a VPlan"' failed.
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.      Program arguments: /var/tmp/clang_15.0.0/RA_3779/bin/clang-15 -cc1 -triple x86_64-unknown-linux-gnu -emit-obj --mrelax-relocations -disable-free -clear-ast-before-backend -main-file-name gen_06734.cpp -mrelocation-model static -mframe-pointer=none -fmath-errno -ffp-contract=on -fno-rounding-math -mconstructor-aliases -funwind-tables=2 -target-cpu x86-64 -tune-cpu generic -mllvm -treat-scalable-fixed-error-as-warning -debugger-tuning=gdb -fcoverage-compilation-dir=/home/tatyana/work/cpp/run_long_clang_15.0.0_fd_3779_9_2/tests -resource-dir /var/tmp/clang_15.0.0/RA_3779/lib/clang/15.0.0 -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0 -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/x86_64-linux-gnu/c++/7.5.0 -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/backward -internal-isystem /var/tmp/clang_15.0.0/RA_3779/lib/clang/15.0.0/include -internal-isystem /usr/local/include -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../x86_64-linux-gnu/include -internal-externc-isystem /usr/include/x86_64-linux-gnu -internal-externc-isystem /include -internal-externc-isystem /usr/include -O3 -fdeprecated-macro -fdebug-compilation-dir=/home/tatyana/work/cpp/run_long_clang_15.0.0_fd_3779_9_2/tests -ferror-limit 19 -fgnuc-version=4.2.1 -fcxx-exceptions -fexceptions -vectorize-loops -vectorize-slp -faddrsig -D__GCC_HAVE_DWARF2_CFI_ASM=1 -o /tmp/gen_06734-678ecb.o -x c++ ./gen_06734.cpp
1.      <eof> parser at end of file
2.      Optimizer
 #0 0x00007fe1d6aa3c2c llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) /home/dbakunevich/work/llvm-project/llvm-project/llvm/lib/Support/Unix/Signals.inc:569:0
 #1 0x00007fe1d6aa1c24 llvm::sys::RunSignalHandlers() /home/dbakunevich/work/llvm-project/llvm-project/llvm/lib/Support/Signals.cpp:103:0
 #2 0x00007fe1d6aa2393 SignalHandler(int) /home/dbakunevich/work/llvm-project/llvm-project/llvm/lib/Support/Unix/Signals.inc:397:0
 #3 0x00007fe1e2bea980 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x12980)
 #4 0x00007fe1d4e7be87 raise (/lib/x86_64-linux-gnu/libc.so.6+0x3ee87)
 #5 0x00007fe1d4e7d7f1 abort (/lib/x86_64-linux-gnu/libc.so.6+0x407f1)
 #6 0x00007fe1d4e6d3fa (/lib/x86_64-linux-gnu/libc.so.6+0x303fa)
 #7 0x00007fe1d4e6d472 (/lib/x86_64-linux-gnu/libc.so.6+0x30472)
 #8 0x00007fe1d7f35409 bool llvm::isa<llvm::VPReplicateRecipe, llvm::VPValue*>(llvm::VPValue* const&) /home/dbakunevich/work/llvm-project/llvm-project/llvm/lib/Transforms/Vectorize/VPlanValue.h:188:0
 #9 0x00007fe1d7f35409 llvm::cast_retty<llvm::VPReplicateRecipe, llvm::VPValue*>::ret_type llvm::dyn_cast<llvm::VPReplicateRecipe, llvm::VPValue>(llvm::VPValue*) /home/dbakunevich/work/llvm-project/llvm-project/llvm/include/llvm/Support/Casting.h:345:0
#10 0x00007fe1d7f35409 llvm::VPTransformState::get(llvm::VPValue*, unsigned int) /home/dbakunevich/work/llvm-project/llvm-project/llvm/lib/Transforms/Vectorize/LoopVectorize.cpp:10100:0
#11 0x00007fe1d7f3905f llvm::InnerLoopVectorizer::widenCallInstruction(llvm::CallInst&, llvm::VPValue*, llvm::VPUser&, llvm::VPTransformState&) /home/dbakunevich/work/llvm-project/llvm-project/llvm/lib/Transforms/Vectorize/LoopVectorize.cpp:4246:0
#12 0x00007fe1d7fcdd64 llvm::ilist_node_impl<llvm::ilist_detail::node_options<llvm::VPRecipeBase, true, false, void> >::getNext() /home/dbakunevich/work/llvm-project/llvm-project/llvm/include/llvm/ADT/ilist_node.h:67:0                   #13 0x00007fe1d7fcdd64 llvm::ilist_iterator<llvm::ilist_detail::node_options<llvm::VPRecipeBase, true, false, void>, false, false>::operator++() /home/dbakunevich/work/llvm-project/llvm-project/llvm/include/llvm/ADT/ilist_iterator.h:157:0                                                                                                                                                                                                                                            #14 0x00007fe1d7fcdd64 llvm::VPBasicBlock::execute(llvm::VPTransformState*) /home/dbakunevich/work/llvm-project/llvm-project/llvm/lib/Transforms/Vectorize/VPlan.cpp:346:0
#15 0x00007fe1d7fc4169 llvm::VPRegionBlock::execute(llvm::VPTransformState*) /home/dbakunevich/work/llvm-project/llvm-project/llvm/lib/Transforms/Vectorize/VPlan.cpp:492:0
#16 0x00007fe1d7fc99d5 llvm::df_iterator<llvm::VPBlockBase*, llvm::df_iterator_default_set<llvm::VPBlockBase*, 8u>, false, llvm::GraphTraits<llvm::VPBlockBase*> >::operator++() /home/dbakunevich/work/llvm-project/llvm-project/llvm/include/llvm/ADT/DepthFirstIterator.h:178:0
#17 0x00007fe1d7fc99d5 llvm::VPlan::execute(llvm::VPTransformState*) /home/dbakunevich/work/llvm-project/llvm-project/llvm/lib/Transforms/Vectorize/VPlan.cpp:949:0
#18 0x00007fe1d7f5868f llvm::LoopVectorizationPlanner::executePlan(llvm::ElementCount, unsigned int, llvm::VPlan&, llvm::InnerLoopVectorizer&, llvm::DominatorTree*) /home/dbakunevich/work/llvm-project/llvm-project/llvm/lib/Transforms/Vectorize/LoopVectorize.cpp:7629:0
#19 0x00007fe1d7f65731 llvm::LoopVectorizePass::processLoop(llvm::Loop*) /home/dbakunevich/work/llvm-project/llvm-project/llvm/lib/Transforms/Vectorize/LoopVectorize.cpp:10605:0
#20 0x00007fe1d7f65ca9 llvm::LoopVectorizePass::runImpl(llvm::Function&, llvm::ScalarEvolution&, llvm::LoopInfo&, llvm::TargetTransformInfo&, llvm::DominatorTree&, llvm::BlockFrequencyInfo&, llvm::TargetLibraryInfo*, llvm::DemandedBits&, llvm::AAResults&, llvm::AssumptionCache&, std::function<llvm::LoopAccessInfo const& (llvm::Loop&)>&, llvm::OptimizationRemarkEmitter&, llvm::ProfileSummaryInfo*) /home/dbakunevich/work/llvm-project/llvm-project/llvm/lib/Transforms/Vectorize/LoopVectorize.cpp:10703:0
#21 0x00007fe1d7f66236 llvm::LoopVectorizePass::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) /home/dbakunevich/work/llvm-project/llvm-project/llvm/lib/Transforms/Vectorize/LoopVectorize.cpp:10735:0
#22 0x00007fe1d9f1af0d llvm::detail::PassModel<llvm::Function, llvm::LoopVectorizePass, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Function> >::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) /home/dbakunevich/work/llvm-project/llvm-project/llvm/include/llvm/IR/PassManagerInternal.h:89:0
#23 0x00007fe1d6c3c4e1 llvm::SmallPtrSet<void*, 2u>::operator=(llvm::SmallPtrSet<void*, 2u>&&) /home/dbakunevich/work/llvm-project/llvm-project/llvm/include/llvm/ADT/SmallPtrSet.h:489:0
#24 0x00007fe1d6c3c4e1 llvm::PreservedAnalyses::operator=(llvm::PreservedAnalyses&&) /home/dbakunevich/work/llvm-project/llvm-project/llvm/include/llvm/IR/PassManager.h:152:0
#25 0x00007fe1d6c3c4e1 llvm::PassManager<llvm::Function, llvm::AnalysisManager<llvm::Function> >::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) /home/dbakunevich/work/llvm-project/llvm-project/llvm/include/llvm/IR/PassManager.h:522:0
#26 0x00007fe1d8e5999d llvm::detail::PassModel<llvm::Function, llvm::PassManager<llvm::Function, llvm::AnalysisManager<llvm::Function> >, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Function> >::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) /home/dbakunevich/work/llvm-project/llvm-project/llvm/include/llvm/IR/PassManagerInternal.h:89:0
#27 0x00007fe1d6c3bcf3 llvm::SmallPtrSet<void*, 2u>::operator=(llvm::SmallPtrSet<void*, 2u>&&) /home/dbakunevich/work/llvm-project/llvm-project/llvm/include/llvm/ADT/SmallPtrSet.h:489:0
#28 0x00007fe1d6c3bcf3 llvm::PreservedAnalyses::operator=(llvm::PreservedAnalyses&&) /home/dbakunevich/work/llvm-project/llvm-project/llvm/include/llvm/IR/PassManager.h:152:0
#29 0x00007fe1d6c3bcf3 llvm::ModuleToFunctionPassAdaptor::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) /home/dbakunevich/work/llvm-project/llvm-project/llvm/lib/IR/PassManager.cpp:127:0
#30 0x00007fe1d8e5a07d llvm::detail::PassModel<llvm::Module, llvm::ModuleToFunctionPassAdaptor, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Module> >::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) /home/dbakunevich/work/llvm-project/llvm-project/llvm/include/llvm/IR/PassManagerInternal.h:89:0
#31 0x00007fe1d6c3a3b7 llvm::SmallPtrSet<void*, 2u>::operator=(llvm::SmallPtrSet<void*, 2u>&&) /home/dbakunevich/work/llvm-project/llvm-project/llvm/include/llvm/ADT/SmallPtrSet.h:489:0
#32 0x00007fe1d6c3a3b7 llvm::PreservedAnalyses::operator=(llvm::PreservedAnalyses&&) /home/dbakunevich/work/llvm-project/llvm-project/llvm/include/llvm/IR/PassManager.h:152:0
#33 0x00007fe1d6c3a3b7 llvm::PassManager<llvm::Module, llvm::AnalysisManager<llvm::Module> >::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) /home/dbakunevich/work/llvm-project/llvm-project/llvm/include/llvm/IR/PassManager.h:522:0
#34 0x00007fe1dfe130a2 llvm::SmallPtrSetImplBase::~SmallPtrSetImplBase() /home/dbakunevich/work/llvm-project/llvm-project/llvm/include/llvm/ADT/SmallPtrSet.h:83:0
#35 0x00007fe1dfe130a2 llvm::SmallPtrSetImpl<llvm::AnalysisKey*>::~SmallPtrSetImpl() /home/dbakunevich/work/llvm-project/llvm-project/llvm/include/llvm/ADT/SmallPtrSet.h:344:0
#36 0x00007fe1dfe130a2 llvm::SmallPtrSet<llvm::AnalysisKey*, 2u>::~SmallPtrSet() /home/dbakunevich/work/llvm-project/llvm-project/llvm/include/llvm/ADT/SmallPtrSet.h:450:0
#37 0x00007fe1dfe130a2 llvm::PreservedAnalyses::~PreservedAnalyses() /home/dbakunevich/work/llvm-project/llvm-project/llvm/include/llvm/IR/PassManager.h:152:0
#38 0x00007fe1dfe130a2 (anonymous namespace)::EmitAssemblyHelper::RunOptimizationPipeline(clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream> >&, std::unique_ptr<llvm::ToolOutputFile, std::default_delete<llvm::ToolOutputFile> >&) (.constprop.839) /home/dbakunevich/work/llvm-project/llvm-project/clang/lib/CodeGen/BackendUtil.cpp:1505:0
#39 0x00007fe1dfe16178 RunCodegenPipeline /home/dbakunevich/work/llvm-project/llvm-project/clang/lib/CodeGen/BackendUtil.cpp:1515:0
#40 0x00007fe1dfe16178 EmitAssembly /home/dbakunevich/work/llvm-project/llvm-project/clang/lib/CodeGen/BackendUtil.cpp:1571:0
#41 0x00007fe1dfe16178 clang::EmitBackendOutput(clang::DiagnosticsEngine&, clang::HeaderSearchOptions const&, clang::CodeGenOptions const&, clang::TargetOptions const&, clang::LangOptions const&, llvm::StringRef, llvm::Module*, clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream> >) /home/dbakunevich/work/llvm-project/llvm-project/clang/lib/CodeGen/BackendUtil.cpp:1734:0
#42 0x00007fe1e018e6e0 std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream> >::~unique_ptr() /home/buildmaster/sw/centaurea/gcc-7.4.0-rhel6/include/c++/7.4.0/bits/unique_ptr.h:267:0
#43 0x00007fe1e018e6e0 clang::BackendConsumer::HandleTranslationUnit(clang::ASTContext&) /home/dbakunevich/work/llvm-project/llvm-project/clang/lib/CodeGen/CodeGenAction.cpp:370:0
#44 0x00007fe1deedc399 clang::ParseAST(clang::Sema&, bool, bool) /home/dbakunevich/work/llvm-project/llvm-project/clang/lib/Parse/ParseAST.cpp:175:0
#45 0x00007fe1e018e9f0 clang::CodeGenAction::ExecuteAction() /home/dbakunevich/work/llvm-project/llvm-project/clang/lib/CodeGen/CodeGenAction.cpp:1110:0
#46 0x00007fe1e0ac2a11 clang::FrontendAction::Execute() /home/dbakunevich/work/llvm-project/llvm-project/clang/lib/Frontend/FrontendAction.cpp:1032:0
#47 0x00007fe1e0a5c500 llvm::Error::setChecked(bool) /home/dbakunevich/work/llvm-project/llvm-project/llvm/include/llvm/Support/Error.h:302:0
#48 0x00007fe1e0a5c500 llvm::Error::operator bool() /home/dbakunevich/work/llvm-project/llvm-project/llvm/include/llvm/Support/Error.h:234:0
#49 0x00007fe1e0a5c500 clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) /home/dbakunevich/work/llvm-project/llvm-project/clang/lib/Frontend/CompilerInstance.cpp:1036:0
#50 0x00007fe1e0b42983 clang::ExecuteCompilerInvocation(clang::CompilerInstance*) /home/dbakunevich/work/llvm-project/llvm-project/clang/lib/FrontendTool/ExecuteCompilerInvocation.cpp:267:0
#51 0x0000000000415e51 cc1_main(llvm::ArrayRef<char const*>, char const*, void*) /home/dbakunevich/work/llvm-project/llvm-project/clang/tools/driver/cc1_main.cpp:248:0
#52 0x0000000000412771 ExecuteCC1Tool(llvm::SmallVectorImpl<char const*>&) /home/dbakunevich/work/llvm-project/llvm-project/clang/tools/driver/driver.cpp:317:0
#53 0x000000000040f259 main /home/dbakunevich/work/llvm-project/llvm-project/clang/tools/driver/driver.cpp:388:0
#54 0x00007fe1d4e5ec87 __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x21c87)
#55 0x0000000000411fea _start (/var/tmp/clang_15.0.0/RA_3779/bin/clang-15+0x411fea)
clang-15: error: unable to execute command: Aborted (core dumped)
clang-15: error: clang frontend command failed due to signal (use -v to see invocation)
clang version 15.0.0 (https://github.com/llvm/llvm-project.git 74b56e02bda09ae735c6aeca583a2078375f2da1)
```

**Comment by dbakunevich** [comment]
This fall was closed due to not being replayable in clang with the assertion trunk.

**Comment by fhahn** [comment]
@dbakunevich thanks for closing the issue! I believe this was fixed by  0ef8ca6d88aa7e4abc11
